js/src/admin/components/BasicsPage.js

```diff
@@ -40,157 +20,109 @@ export default class BasicsPage extends Page {
-                  <textarea className="FormControl" bidi={this.values.welcome_message} />
-                </div>,
-              ]
-            )}
-
-            {Object.keys(this.displayNameOptions).length > 1 ? (
-              <FieldSet label={app.translator.trans('core.admin.basics.display_name_heading')}>
-                <div className="helpText">{app.translator.trans('core.admin.basics.display_name_text')}</div>
-                <Select
-                  options={this.displayNameOptions}
-                  value={this.values.display_name_driver()}
-                  onchange={this.values.display_name_driver}
-                ></Select>
-              </FieldSet>
-            ) : (
-              ''
-            )}
-
-            {Object.keys(this.slugDriverOptions).map((model) => {
-              const options = this.slugDriverOptions[model];
-              if (Object.keys(options).length > 1) {
-                return (
-                  <FieldSet label={app.translator.trans('core.admin.basics.slug_driver_heading', { model })}>
-                    <div className="helpText">{app.translator.trans('core.admin.basics.slug_driver_text', { model })}</div>
-                    <Select options={options} value={this.values[`slug_driver_${model}`]()} onchange={this.values[`slug_driver_${model}`]}></Select>
-                  </FieldSet>
-                );
-              }
-            })}
-
-            {Button.component(
-              {
-                type: 'submit',
-                className: 'Button Button--primary',
-                loading: this.loading,
-                disabled: !this.changed(),
-              },
-              app.translator.trans('core.admin.basics.submit_button')
-            )}
-          </form>
+  content() {
+    return [
+      <div className="Form">
+        {this.buildSettingComponent({
+          type: 'text',
+          setting: 'forum_title',
+          label: app.translator.trans('core.admin.basics.forum_title_heading'),
+        })}
+        {this.buildSettingComponent({
+          type: 'text',
+          setting: 'forum_description',
+          label: app.translator.trans('core.admin.basics.forum_description_heading'),
+          help: app.translator.trans('core.admin.basics.forum_description_text'),
+        })}
+
+        {Object.keys(this.localeOptions).length > 1
+          ? [
+              this.buildSettingComponent({
+                type: 'select',
+                setting: 'default_locale',
+                options: this.localeOptions,
+                label: app.translator.trans('core.admin.basics.default_language_heading'),
+              }),
+              this.buildSettingComponent({
+                type: 'switch',
+                setting: 'show_language_selector',
+                label: app.translator.trans('core.admin.basics.show_language_selector_label'),
+              }),
+            ]
+          : ''}
+
+        {FieldSet.component(
```

Comment:
Maybe this one should be JSX?

Reply:
Done, also did the same in the mail page.